lib/taskana-core/src/test/java/pro/taskana/BulkOperationResultsTest.java

```diff
@@ -0,0 +1,98 @@
+package pro.taskana;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Unit test of BulkOperationResults.
+ */
+public class BulkOperationResultsTest {
+
+    private static final Integer ID1 = 1;
+    private static final Integer ID2 = 2;
+    private static final RuntimeException ERROR1 = new RuntimeException("error1");
+    private static final RuntimeException ERROR2 = new RuntimeException("error2");
+
+    BulkOperationResults<Integer, RuntimeException> bulkOperationResults;
+
+    @Before
+    public void setUp() {
+        bulkOperationResults = new BulkOperationResults<Integer, RuntimeException>();
+    }
+
+    @Test
+    public void getErrorMap() {
+        Map<Integer, RuntimeException> map = bulkOperationResults.getErrorMap();
+        assertNotNull(map);
+    }
+
+    @Test
+    public void addError() {
+        bulkOperationResults.addError(ID1, ERROR1);
+        assertEquals(ERROR1, bulkOperationResults.getErrorForId(ID1));
+    }
+
+    @Test
+    public void containsErrors() {
+        assertFalse(bulkOperationResults.containsErrors());
+        bulkOperationResults.addError(ID1, ERROR1);
+        assertTrue(bulkOperationResults.containsErrors());
+    }
+
+    @Test
+    public void getErrorForId() {
+
+        assertNull(bulkOperationResults.getErrorForId(ID1));
+        bulkOperationResults.addError(ID1, ERROR1);
+        assertEquals(ERROR1, bulkOperationResults.getErrorForId(ID1));
+    }
+
+    @Test
+    public void getFailedIds() {
```

Comment:
What I am missing: The list that is returned should not be linked to the internal list. So a change to the returned list should not result in a change within the internal map.